kinto/tests/test_views_collections.py

```diff
@@ -35,12 +35,15 @@ def test_collections_name_should_be_simple(self):
                           headers=self.headers,
                           status=400)
 
-    def test_unknown_bucket_raises_404(self):
+    def test_unknown_bucket_raises_403(self):
```

Comment:
Because you don't know if it exists or if you don't have access to it.
The permission backend doesn't give you a allowed if the the object doesn't exists.
So we should call the storage backend before the permission backend if we wanted a 404 here.

Also most security guys will tell you that it is better to return a 403 in that case.